### DISCUSSION TITLE
GLPI Agent Printer Problem

### DISCUSSION THREAD

**Discussion by CamiloPerezOrtiz** [glpi-project/glpi-agent#420] [Q&A]
### Bug reporting acknowledgment

Yes, I read it

### Professional support

None

### Describe the bug

GLPI Printer Agent Problem
Hello, I hope you can help me, I am working with glpi version 10.0.6, I am currently using glpi-agent 1.4, when I run the agent on a specific Linux computer Linux Mint 20.1 version 20.1 (Ulyssa), it detects all the software, hardware, etc, the only thing that doesn't detect me is the printer, I've checked several github threads but I can't solve it, the printer I'm using is an EPSON EcoTank L3150, I attach a screenshot of the glpi-netinventory –host command

Thanks for the help

![image](https://github.com/glpi-project/glpi-agent/assets/41083269/1d02002d-a0c2-4516-8c73-030555ceb204)

### To reproduce

1.Download agent for linux https://github.com/glpi-project/glpi-agent/releases/download/1.4/glpi-agent-1.4-linux-installer.pl
2. Install and configure the server | server = http://xxxx/front/inventory.php
3. running agent: glpi-agent - glpi-agent --debug

### Expected behavior

Agent recognize the printer

### Operating system

Linux

### GLPI Agent version

1.4

### GLPI version

10.0.6

**Comment by g-bougard** [comment]
Hi @CamiloPerezOrtiz 

as you could have read on [glpi-netinventory man page](https://glpi-agent.readthedocs.io/en/latest/man/glpi-netinventory.html), this script is used for standalone network inventory. And the `--host` option have to be a hostanme or an ip. This has nothing to do with local printer inventory. If your printer has a network interface, you can try to inventory it this way if it supports SNMP and SNMP support is enabled on it.

For local printer inventory you indeed require the `Net::CUPS` perl module. You should check to install via your system package management. For example, on RedHat/CentOS/Fedora, the package is `perl-Net-CUPS` ou Debian, this should be `libnet-cups-perl`. I don't know which package should be installed on Linux Mint, but I hope my explanation will help you to find it.

**Comment by CamiloPerezOrtiz** [comment]
Hi @g-bougard , thanks for clarification.
Install the package that was missing, but even so I cannot inventory the local printer, when I run the glpi-agent command with the debug if the printer appears but on the web it is not inventorying, it even running the glpi-inventory command if the printer appears, am i doing something wrong?

libnet-cups-perl
![image](https://github.com/glpi-project/glpi-agent/assets/41083269/69efe7c1-5f65-48f1-b6b4-8d49fede1c6f)


glpi-agent --debug
![image](https://github.com/glpi-project/glpi-agent/assets/41083269/31e48ce7-1b63-4819-9814-edc268c3243a)

glpi-inventory
![image](https://github.com/glpi-project/glpi-agent/assets/41083269/f005b9fa-4904-49bc-9192-03adc259a99b)

**Reply by g-bougard** [reply]
Hi @CamiloPerezOrtiz 
the problem here is no serial has been found by the agent. Of course, here, we see it can be extracted from the name and even the name could be shorten to just "EPSON/L3150 Series". Without a serialnumber, GLPI won't import it as the default import rules will only import printers with a serial.

**Reply by g-bougard** [reply]
I just added 81292ae commit which should extract the serial from Cups printer uri in your case.
Please give next nightly build a try.

**Reply by CamiloPerezOrtiz** [reply]
Hi @g-bougard 
Thank you very much, I just tried it and it worked correctly, in the Linux Mint Operating System, attached capture, thank you.

![image](https://github.com/glpi-project/glpi-agent/assets/41083269/16cbfec1-2993-444e-bcf9-53521391761c)

I close the discussion.

**Reply by g-bougard** [reply]
Indeed, this is not fully correct. `&interface=1` should not be included in the serial. I think I missed a point. I'll fix it and you probably have to delete this printer when you'll try next nightly or you also can cleanup the serialnumber from the `&interface=1` part.

**Reply by g-bougard** [reply]
I just push 581d15c commit which should fix this last problem.

**Reply by CamiloPerezOrtiz** [reply]
Hi @g-bougard 

Thanks for the comment, I just tried it and if it already brings the serial number right, I attach a screenshot, I appreciate the support.

![image](https://github.com/glpi-project/glpi-agent/assets/41083269/46f3ea48-5054-4085-9b83-57112870fbf2)

As a note, in case someone else has the same detail, it only happened to me with the Linux operating system (Linux Mint), with Windows there was no problem.
